### PR TITLE
Add docker and network pod functions to e2e framework

### PR DESCRIPTION
**Pull request by mkimuram**
This PR is to add docker and network pod functions to e2e framework. These functions are originally needed for https://github.com/submariner-io/submariner/pull/1202 , however, as a result of discussion, we came to conclusion that it would be nice to add them to this framework to make room to leverage from other tests.

### CODE REVIEW

File: test/e2e/framework/docker.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+	// output has trailing "\n", so it needs to be trimed
+	return strings.TrimSuffix(stdout.String(), "\n")
+}
+
+func (d *Docker) GetLog() string {
+	var stderr bytes.Buffer
+	// get stderr of `docker log {d.rame}` command
+	cmd := exec.Command("docker", "logs", d.Name)
+	cmd.Stderr = &stderr
```
Comment (sridhargaddam) [inline test/e2e/framework/docker.go:55]: Why only stderr and not stdout?

Comment (mangelajo) [reply test/e2e/framework/docker.go:55]: :+1:

**Review by mangelajo — COMMENTED** [review]
Looks good, only the sridhar comment.

File: test/e2e/framework/docker.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	"strings"
+
+	. "github.com/onsi/gomega"
+)
+
+type Docker struct {
+	Name string
+}
+
+func NewFromExistingDocker(name string) *Docker {
```
Comment (tpantelis) [inline test/e2e/framework/docker.go]:
```suggestion
func New(dockerName string) *Docker {
```

File: test/e2e/framework/network_pods.go [hunk 3]
```diff
@@ -178,6 +188,61 @@ func (np *NetworkPod) CreateService() *v1.Service {
 	return np.framework.CreateTCPService(np.Config.Cluster, np.Pod.Labels[TestAppLabel], np.Config.Port)
 }
 
+// RunCommand run the specified comand in this NetworkPod
```
Comment (tpantelis) [inline test/e2e/framework/network_pods.go]:
```suggestion
// RunCommand run the specified command in this NetworkPod
```

File: test/e2e/framework/network_pods.go [hunk 4]
```diff
@@ -178,6 +188,61 @@ func (np *NetworkPod) CreateService() *v1.Service {
...
+
+// GetLog returns container log from this NetworkPod
+func (np *NetworkPod) GetLog() string {
+	req := KubeClients[np.Config.Cluster].CoreV1().RESTClient().Get().
+		Namespace(np.Pod.Namespace).Name(np.Pod.Name).Resource("pods").SubResource("log")
+
+	req.Param("follow", strconv.FormatBool(false)).
+		Param("container", np.Config.ContainerName).
+		Param("previous", strconv.FormatBool(false)).
+		Param("timestamps", strconv.FormatBool(false))
```
Comment (tpantelis) [inline test/e2e/framework/network_pods.go]: It seems like you could use 

`KubeClients[np.Config.Cluster].CoreV1().Pods(np.Pod.Namespace).GetLogs(np.Pod.Name, &corev1.PodLogOptions{...})`

File: test/e2e/framework/network_pods.go [hunk 5]
```diff
@@ -404,6 +469,38 @@ func (np *NetworkPod) buildLatencyServerPod() {
...
+			},
+			Tolerations: []v1.Toleration{{Operator: v1.TolerationOpExists}},
+		},
+	}
+
+	pc := KubeClients[np.Config.Cluster].CoreV1().Pods(np.framework.Namespace)
+	var err error
+	np.Pod, err = pc.Create(&customPod)
+	Expect(err).NotTo(HaveOccurred())
+	np.AwaitReady()
```
Comment (tpantelis) [inline test/e2e/framework/network_pods.go]: maybe add some newlines for readability...

```suggestion
	pc := KubeClients[np.Config.Cluster].CoreV1().Pods(np.framework.Namespace)
	
	var err error
	np.Pod, err = pc.Create(&customPod)
	Expect(err).NotTo(HaveOccurred())
	
	np.AwaitReady()
```

**Review by tpantelis — CHANGES_REQUESTED** [review]

File: test/e2e/framework/network_pods.go (hunk 4 above)
Comment (mkimuram) [reply test/e2e/framework/network_pods.go]: Good catch. Actually, I also found that I was specifying default values for `.PodLogOptions` (Even `container` isn't necessary, if there is only one). It now become much simple.

**Comment by mkimuram** [comment]
@mangelajo @sridhargaddam @tpantelis 

Thank you for your reviews. I've fixed and squashed the commit.
I tested locally and confirmed that it works well with the submariner side's PR with "Fix for shipyard framework changes" commit.

PTAL

**Review by tpantelis — APPROVED** [review]

**Review by mangelajo — APPROVED** [review]

**Comment by mangelajo** [comment]
Great work thanks @mkimuram

**Review by sridhargaddam — APPROVED** [review]
Great work @mkimuram

**Pull request merged**